tests/test_metric_regrouping.js

```diff
@@ -273,14 +273,25 @@ test('bundling', function(t) {
       counters = metric_utils.extractCounters(parsedObj),
       timers = metric_utils.extractTimers(parsedObj),
       sets = metric_utils.extractSets(parsedObj),
-      jsonArray = http_client.buildPayloadsUnsafe('11111111', 22222222, gauges, counters, timers, sets, 15000, 200),
+      maxPayloadSize = 600,
+      jsonArray = http_client.buildPayloadsUnsafe('11111111', 22222222, gauges, counters, timers, sets, 15000, maxPayloadSize),
       actualMetricCount = 0;
 
-  // it should have been broken up into 12 bundles.
-  t.equal(14, jsonArray.length, 'The payload was broken into 14 bundles.');
+  // it should have been broken up into 6 bundles.
+  t.equal(6, jsonArray.length, 'The payload was broken into 6 bundles.');
 
   // ensure that we have the expected number of metrics.
   jsonArray.forEach(function(json) {
+    payloadLength = json.length;
+    if(payloadLength == 643){
```

Comment:
Since this is in a test case, is this condition needed? Will `payloadLength` ever not be 643?

Reply:
Yes. There are two test cases that are handled in this logic. One is where all but one are less than or equal to max length of 600, and the other one is that only one case where if a single metric is of bigger size, that gets tested too.

Reply:
Thanks, I see now that it is iterating over that `jsonArray`.